File: LeetCode/Stack/OnlineStockSpan.py
from collections import deque
class StockSpanner:

    def __init__(self):
        self.stack=deque()

    def next(self, price: int) -> int:
        count=1
        while len(self.stack)>0 and self.stack[-1][0]<=price:
            count=count+self.stack[-1][1]
            self.stack.pop()
        self.stack.append((price,count))
        return self.stack[-1][1]
        

File: Stack/Stack-Java/OnlineStockSpan.java

class StockSpanner {
    Stack<int []> st;
    public StockSpanner() {
        st=new Stack<>();
    }
    
    public int next(int price) {
     int span =1; 
      while(st.size()>0 && st.peek()[0]<=price){
          span+=st.pop()[1];
      }
        st.push(new int[]{price,span});
      return span;
    }
}

//leetcode link : https://leetcode.com/problems/online-stock-span/submissions/
//tc:o(n)

//sc:o(n)

//refer to picture in this solution for better understanding: https://leetcode.com/problems/online-stock-span/discuss/640648/Clean-Java-Stack-solution-with-explanation.-90-better-run-time
